cpp/lib/src/Zotero.cc

```diff
@@ -0,0 +1,86 @@
+/** \brief Interaction with Zotero Translation Server
+ *         public functions are named like endpoints
+ *         see https://github.com/zotero/translation-server
+ *  \author Mario Trojan
+ *
+ *  \copyright 2018 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "Zotero.h"
+#include <uuid/uuid.h>
+#include "JSON.h"
+#include "StringUtil.h"
+#include "util.h"
+
+
+const std::vector<std::string> Zotero::ExportFormats(
+    { "bibtex", "biblatex", "bookmarks", "coins", "csljson", "mods", "refer",
+    "rdf_bibliontology", "rdf_dc", "rdf_zotero", "ris", "wikipedia", "tei" }
+);
+
+
+// We try to be unique for the machine we're on.  Beyond that we may have a problem.
+std::string Zotero::GetNextSessionId() {
+    static unsigned counter;
+    static uint32_t uuid[4];
+    if (unlikely(counter == 0))
+        ::uuid_generate(reinterpret_cast<unsigned char *>(uuid));
+    ++counter;
+    return "ub_tools_zts_client_" + StringUtil::ToString(uuid[0]) + StringUtil::ToString(uuid[1])
```

Comment:
Why do you only use 3 of the 4 values of uuid?

Reply:
blindness in the morning? caffeine hasn't kicked in yet? ;-)